### ISSUE TITLE
Command line "Save to Folder" only outputs database.json, no folders

### ISSUE THREAD

**Issue by BVMather** [otykier/TabularEditor#652]
When I choose Save to Folder in the Tabular Editor is works as expected, I get all the folders (cultures, perspectives, relationships, roles, tables) and all the relevant files and folders nested underneath.

If I try the same using the command line with -F:

    $cmd = "& ""C:\Program Files (x86)\Tabular Editor\TabularEditor.exe"" ""$Server"" ""$DatabaseName"" -F ""$ModelFolderPath""  "

then I only get the database.json file and nothing else.

**Comment by otykier** [comment]
When you use Tabular Editor's UI to save a model as a folder structure, Tabular Editor uses the serialization specified under File > Preferences > Serialization:

<img src="https://user-images.githubusercontent.com/8976200/96698570-91101980-138d-11eb-9029-bdf2473531d2.png" width="300" />

If the model holds an annotation that specifies the serialization settings, there is a checkbox at the lower right corner of the dialog, where you can decide whether to use the annotated settings, or (if the checkbox is unchecked) to use the settings specified above:

<img src="https://user-images.githubusercontent.com/8976200/96698871-e6e4c180-138d-11eb-8362-fa5947cf7216.png" width="500" />

When running Tabular Editor from the CLI, it will always use the serialization settings specified in the file. Check that the model you want to save has the `TabularEditor_SerializeOptions` annotation specified at the model level. If there is none, or if the settings in the annotation do not specify individual object types for serialization in the `Levels` json array, then the `-F` switch will cause all the metadata to be saved in a single database.json file. 

If you're not able to include or modify the annotation on the server where your model is deployed, you can add the annotation on-the-fly using the CLI, before the model is saved to folder. This is done using the `-S` switch.

To do this, create a .json file containing the serialization settings you want to use:

**TabularEditor_SerializeOptions.json**
```json
{
  "IgnoreInferredObjects": true,
  "IgnoreInferredProperties": true,
  "IgnoreTimestamps": true,
  "SplitMultilineStrings": true,
  "PrefixFilenames": false,
  "LocalTranslations": false,
  "LocalPerspectives": false,
  "LocalRelationships": false,
  "Levels": [
    "Data Sources",
    "Perspectives",
    "Relationships",
    "Roles",
    "Tables",
    "Tables/Columns",
    "Tables/Measures",
    "Translations"
  ]
}
```

Also, create a script that loads this file into an annotation on the model:

**ApplySerializeOptionsAnnotation.csx**
```
Model.SetAnnotation("TabularEditor_SerializeOptions", ReadFile("TabularEditor_SerializeOptions.json"));
```

Then, modify your command line to execute this script prior to saving the model:

$cmd = "& ""C:\Program Files (x86)\Tabular Editor\TabularEditor.exe"" ""$Server"" ""$DatabaseName"" -S ""ApplySerializeOptionsAnnotation.csx"" -F ""$ModelFolderPath""  "

**Comment by BVMather** [comment]
Thanks Daniel. 

If I try to run TE from the command line including the script -S ""ApplySerializeOptionsAnnotation.csx""  -F ""$ModelFolderPath"" it doesn't seem to work, it still just outputs database.json without deserializing. I am running it against an open Power BI Desktop localhost.

However, it works as expected (deserializing to folders) if I run the script:

`Model.SetAnnotation("TabularEditor_SerializeOptions", ReadFile("TabularEditor_SerializeOptions.json")); `

directly against the model in Tabular Editor first (and save the changes to the connected database).  It adds the annotation to the PBI desktop model and the annotation appears to persist if the Power BI Desktop file is saved and reopend.

Thanks for your help.

**Comment by otykier** [comment]
Fixed in [2.13.0](https://github.com/otykier/TabularEditor/releases/latest)

**Comment by Tim-McKenzie** [comment]
Hi Daniel,

I have encountered this challenge using the CLI to serialize model to folder.
Using 2.15.2 (so above fix introduced in 2.13.0 should be in place)

I am hitting the same aspect BVMather documented on 3 Nov 2020.

If I run the annotation script directly on the model, and then save to connected DB - And then run the CLI to serialize it exports as expected.
However, if I run this fully scripted - the annotation doesn't persist - and hence the folder serialization only outputs the database.json without any folder serialization. 
BTW - Running from PBI Desktop, and not a bim file. Initially I thought this might be to do with ensuring experimental features is turned on, however that doesn't seem to resolve this for me. 

Hopefully this is just user-error, or potentially, in 2.15.2 this issue has cropped back up.

Thanks,
Tim.

**Comment by otykier** [comment]
@Tim-McKenzie 
Just tested this with 2.16.5, and I do not see the issue you outline. Are you sure the script gets executed correctly? You should see the following information in the CLI output:

<img width="683" alt="image" src="https://user-images.githubusercontent.com/8976200/150956415-1a7752d4-cb7e-4134-97d9-e7ffeeff9e8e.png">

Folder structure after running the CLI:

![image](https://user-images.githubusercontent.com/8976200/150956530-5dca777c-372f-4aec-a025-1b573a56896d.png)

Note that I'm running this against a local instance of SQL Server Analysis Services, but that should be the same as running it against PBI Desktop. One thing to be aware about, is that you may need to fully specify the path of the .csx script file in the CLI.
